### PR TITLE
patching dtype for int32 and bool casting

### PR DESCRIPTION
**Pull request by jjsjann123**
Fixes issues exposed in OpInfoTest https://github.com/pytorch/pytorch/pull/71299

This fixes a few op failures, which I didn't add comprehensive tests for. The idea is at some point we'll have OpInfoTests enabled and would run through those for extensive tests.

### CODE REVIEW

**Comment by jjsjann123** [comment]
Good for review. I'm running tests locally

File: torch/csrc/jit/codegen/cuda/arith.cpp [hunk 1]
```diff
@@ -730,6 +731,7 @@ TensorView* min(
       init = IrBuilder::create<Double>(FLT_MAX);
       break;
     case (DataType::Int):
+    case (DataType::Int32):
```
Comment (naoyam) [inline torch/csrc/jit/codegen/cuda/arith.cpp:738]: This is not about your new code, but if `DataType::Int` is 64-bit, shouldn't `init` be `2^64 -1`? I believe `INT_MAX` is 32-bit max.

Comment (jjsjann123) [reply torch/csrc/jit/codegen/cuda/arith.cpp:738]: That's true and I actually was surprised to find that we are using INT_MAX/INT_MIN for those.
I'll update those to LONG_MIN & LONG_MAX.

Comment (naoyam) [reply torch/csrc/jit/codegen/cuda/arith.cpp:738]: `LONG_MAX` is the same as `INT_MAX` on systems where `long` is 32-bit. I vaguely remember that's the case on Windows.

`std::numeric_limits<int64_t>::max` should work.

Comment (jjsjann123) [reply torch/csrc/jit/codegen/cuda/arith.cpp:738]: std::numeric_limits needs to include <limits>, would that give us issue in nvrtc?

Comment (naoyam) [reply torch/csrc/jit/codegen/cuda/arith.cpp:738]: This file is not compiled by nvrtc, so isn't it just fine?

Comment (jjsjann123) [reply torch/csrc/jit/codegen/cuda/arith.cpp:738]: NVM, I was stupid and was cross referencing something else in a different PR :stuck_out_tongue:

File: torch/csrc/jit/codegen/cuda/runtime/helpers.cu [hunk 2]
```diff
@@ -205,6 +205,18 @@ __device__ int64_t where(bool c, int64_t a, int64_t b) {
   return c ? a : b;
 }
 
+__device__ int64_t where(bool c, int a, int b) {
```
Comment (naoyam) [inline torch/csrc/jit/codegen/cuda/runtime/helpers.cu]: Shouldn't the return type be `int`?

Comment (jjsjann123) [reply torch/csrc/jit/codegen/cuda/runtime/helpers.cu]: Oops, my bad... I blame copy-paste

**Review by naoyam — APPROVED** [review]
LGTM

**Comment by jjsjann123** [comment]
Rerunning tests. I'll merge afterwards.

**Comment by jjsjann123** [comment]
Test passing after reverting `Revert "Extend use of SimplyfingIrBuilder (#1448)"` merging

**Pull request merged**